apps/namada-interface/src/App/Staking/ValidatorDetails/ValidatorDetails.tsx

```diff
@@ -94,20 +98,98 @@ const validatorToDataRows = (
   ];
 };
 
+type StakingViewProps = {
+  onRequestClose: (modalOnRequestCloseType: ModalOnRequestCloseType) => void;
+};
+
+const StakingView = (props: StakingViewProps): JSX.Element => {
+  const { onRequestClose } = props;
+  return (
+    <>
+      <Button
+        variant={ButtonVariant.Contained}
+        onClick={() => onRequestClose(ModalOnRequestCloseType.Confirm)}
+      >
+        Confirm
+      </Button>
+      <Button
+        variant={ButtonVariant.Contained}
+        onClick={() => onRequestClose(ModalOnRequestCloseType.Cancel)}
+        style={{ backgroundColor: "lightgrey", color: "black" }}
+      >
+        Cancel
+      </Button>
+    </>
+  );
+};
+
+enum ModalState {
+  None,
+  Stake,
+  Unstake,
+}
+
+enum ModalOnRequestCloseType {
+  Confirm,
+  Cancel,
+}
+
 export const ValidatorDetails = (props: Props): JSX.Element => {
   const { validator, stakingPositionsWithSelectedValidator = [] } = props;
   const validatorDetailsData = validatorToDataRows(validator);
 
+  const [modalState, setModalState] = useState(ModalState.None);
+
+  const onRequestCloseStakingModal = (
+    modalOnRequestCloseType: ModalOnRequestCloseType
+  ): void => {
+    switch (modalOnRequestCloseType) {
+      case ModalOnRequestCloseType.Confirm: {
+        console.log(`called action ${modalOnRequestCloseType}`);
+        setModalState(ModalState.None);
+        break;
+      }
+      case ModalOnRequestCloseType.Cancel: {
+        console.log(`called action ${modalOnRequestCloseType}`);
+        setModalState(ModalState.None);
+        break;
+      }
+    }
+  };
+
+  const myStakingWithValidatorConfigurations =
+    getMyStakingWithValidatorConfigurations(setModalState);
+
   return (
     <ValidatorDetailsContainer>
+      <Modal
+        isOpen={modalState === ModalState.Stake}
+        title="Stake"
+        onBackdropClick={() => {
```

Comment:
It would be cool in the future to bind the `Esc` key as well, as that is pretty universal among modals, but not critical at this point.